src/plugins/intel_gpu/src/plugin/infer_request.cpp

```diff
@@ -918,7 +918,7 @@ void InferRequest::prepare_input(const cldnn::primitive_id& inputName, Blob::Ptr
             _deviceInputs[inputName] = reinterpret_device_blob(_deviceInputs[inputName], inputBlob->getTensorDesc());
         }
     } else if (input_layout.is_static() && !is_dev_input && can_use_usm) {
-        allocate_dev_mem_if_needed(_deviceInputs, inputBlob, inputName, input_layout);
+        allocate_dev_mem_if_needed(_deviceInputs, inputBlob, inputName, input_layout, true);
```

Comment:
You enforced need_lockable_memory to true, which means that we may have copy from system memory to usm *host* memory which will be assigned to first kernel then which is inefficient.

Reply:
Because in the case of unsupported precisions, data must be converted to supported precisions using a write lock as below, I updated to get lockable memory only for the cases. Thank you.

https://github.com/openvinotoolkit/openvino/blob/fac45d79e4d2433e090f2a7d185fa990810eb74c/src/plugins/intel_gpu/src/plugin/infer_request.cpp#L958-L970

Reply:
How about replace it by adding an reorder kernel ?

Reply:
The purpose of `prepare_input()` is to make the input data accessible from GPU. So, the conversion logic is executed only when `is_dev_input` is false, and if the input data is located in a location that cannot be accessed from GPU, such as a host buffer, it is not possible to process the conversion logic by a reorder kernel on the GPU device.

Reply:
>>  if the input data is located in a location that cannot be accessed from GPU, such as a host buffer,

What I meant was, how about add a new first reorder which gets cl_mem and writes the output to usm_device ?

Reply:
@vladimir-paramuzov  I have an additional question, is there an essential issue that the gpu plugin can't support I16/F64/I64 inputs? If not, I am just curious that whether they can be converted just adding a reorder to the network. (Even thought not to be done right now, but want to know the ultimate reason if any. :)

Reply:
@yeonbok We just didn't have support of such data types in cldnn historically, so we can extend graph and reorder kernels to support more data types indeed. I think it's pretty easy for integer types, but with fp64 we may face an issue that it requires cl_khr_fp64 extension support by a device, and as I can see it's not supported by iGPUs at least. So we still need a custom conversion code.

Reply:
@vladimir-paramuzov  Thank you for the explanation of the history. Could I ask one more thing? In this portion, the type conversion is done only when is_dev_input is false. If is_dev_input is true, how those precisions are supported?

Reply:
@yeonbok good question.. Looks like a possible bug in the impl. I think we need to add tests for such cases and ensure that those work.

Reply:
@vladimir-paramuzov  Thanks for the response. Then, how about 
1) Add the reorder strategy in a backlog & do it in the near future 
2) Add the exception & additional test for that case  in this PR? 
 @e-ddykim  @vladimir-paramuzov

Reply:
I'm not clear what problem we're trying to solve, whether it's one problem or multiple.
After finishing this PR, how about clarifying what's next? @vladimir-paramuzov @yeonbok

Reply:
Since the fix itself for the original issues is merged to release branch, so now, I was seeing this PR as a fix for the missing hole of the original flow. So, it would be great if the observed missing part can be checked with simple work so that we can prevent future error. But of course, that can be separately done according to your priority and the required task to be done for the suggestion. Let me create a new backlog conaining both suggestion.

Reply:
I'd like to add one more observation : There is an issue in remote device mem + unsupported precisions. 
 ./benchmark_app -d GPU.1 -m ./models/~~~.xml  -ip I16 

[GPU] shared buffer has smaller size (768) than specified layout (1536)

Anyway I will create it as an issue.